network-config-analyzer/NetworkConfig.py

```diff
@@ -57,59 +51,55 @@ def layer_name_to_config_type(layer_name):
                 return NetworkConfig.ConfigType.Ingress
             return None
 
-    def __init__(self, name, peer_container, policies_container, config_type=None):
+        def config_type_to_layer(self):
+            if self == NetworkConfig.ConfigType.K8s:
+                return NetworkLayerName.K8s_Calico
+            elif self == NetworkConfig.ConfigType.Calico:
+                return NetworkLayerName.K8s_Calico
+            elif self == NetworkConfig.ConfigType.Istio:
+                return NetworkLayerName.Istio
+            elif self == NetworkConfig.ConfigType.Ingress:
+                return NetworkLayerName.Ingress
+            return None
+
+    def __init__(self, name, peer_container, policies_container):
         """
         :param str name: A name for this config
         :param PeerContainer peer_container: The set of endpoints and their namespaces
-        :param PoliciesContainer policies_container : The container of policies, profiles,
-        and sorted policies
-        :param ConfigType config_type: The type of configuration
         """
         self.name = name
         self.peer_container = peer_container
-        self.policies = policies_container.policies or {}
-        self.sorted_policies = policies_container.sorted_policies or []
-        self.ingress_deny_policies = policies_container.ingress_deny_policies or []
-        self.profiles = policies_container.profiles or {}
-        self.allowed_labels = set()
-        if self.policies:
-            self._set_allowed_labels()
+        self.policies = policies_container.policies or {}  # map from policy name to policy object
+        self.layers = policies_container.layers or NetworkLayersContainer()  # map from layer name to layer object
```

Comment:
Why NetworkConfig doesn't have a PoliciesContainer member instead of these two members?
This may simplify things

Reply:
done